examples/model_selection/plot_grid_search_text_feature_extraction.py

```diff
@@ -0,0 +1,235 @@
+:ref:`sphx_glr_auto_examples_text_plot_document_classification_20newsgroups.py`
+notebook.
+
+"""
+
+# Author: Olivier Grisel <olivier.grisel@ensta.org>
+#         Peter Prettenhofer <peter.prettenhofer@gmail.com>
+#         Mathieu Blondel <mathieu@mblondel.org>
+#         Arturo Amor <david-arturo.amor-quiroz@inria.fr>
+# License: BSD 3 clause
+
+# %%
+# Data loading
+# ------------
+# We load two categories from the training set. You can adjust the number of
+# categories by adding their names to the list or setting `categories=None` in
+# the dataset loader to get the 20 of them.
+
+from sklearn.datasets import fetch_20newsgroups
+
+categories = [
+    "alt.atheism",
+    "talk.religion.misc",
+]
+
+data_train = fetch_20newsgroups(
+    subset="train",
+    categories=categories,
+    shuffle=True,
+    random_state=42,
+    remove=("headers", "footers", "quotes"),
+)
+
+data_test = fetch_20newsgroups(
+    subset="test",
+    categories=categories,
+    shuffle=True,
+    random_state=42,
+    remove=("headers", "footers", "quotes"),
+)
+
+print(f"Loading 20 newsgroups dataset for {len(data_train.target_names)} categories:")
+print(data_train.target_names)
+print(f"{len(data_train.data)} documents")
+
+# %%
+# Pipeline with hyperparameter tuning
+# -----------------------------------
+# We define a pipeline combining a text feature vectorizer with a simple
+# classifier.
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.naive_bayes import ComplementNB
+from sklearn.pipeline import Pipeline
+
+pipeline = Pipeline(
+    [
+        ("vect", TfidfVectorizer()),
+        ("clf", ComplementNB()),
+    ]
+)
+pipeline
+
+# %%
+# We define a grid of hyperparameters to be explored by the
+# :class:`~sklearn.model_selection.RandomizedSearchCV`. Using a
+# :class:`~sklearn.model_selection.GridSearchCV` instead would explore all the
+# possible combinations on the grid, which can be costly to compute, whereas the
+# parameter `n_iter` of the :class:`~sklearn.model_selection.RandomizedSearchCV`
+# controls the number of different random combination that are evaluated. Notice
+# that setting `n_iter` larger than the number of possible combinations in a
+# grid would lead to repeating already-explored combinations.
+
+import numpy as np
+
+parameters = {
+    "vect__max_df": (0.2, 0.4, 0.6, 0.8, 1.0),
+    "vect__min_df": (1, 3, 5, 10),
+    "vect__ngram_range": ((1, 1), (1, 2)),  # unigrams or bigrams
+    "vect__norm": ("l1", "l2"),
+    "clf__alpha": np.logspace(-6, 6, 13),
+}
+
+# %%
+# We search for the best parameters for both the feature extraction and the
+# classifier. In this case `n_iter=40` is not an exhaustive search of the
+# hyperparameter grid. In practice it would be interesting to increase the
+# parameter `n_iter` to get a more informative analysis. The consequent increase
+# in computing time can be handled by increasing the number of CPUs via the
+# `n_jobs` parameter.
```

Comment:
This suggests:
 - moving the second paragraph's first sentence to the last pargraph
 - adding indications regarding parameters' by mentioning prefixes in 
 this sentence 
 - a typo fix ("hyperparameters' grid")

```suggestion
# grid would lead to repeating already-explored combinations. We search for
# the best parameters for both the feature extraction (`vect__`) and the
# classifier (`clf__`).

import numpy as np

parameters = {
    "vect__max_df": (0.2, 0.4, 0.6, 0.8, 1.0),
    "vect__min_df": (1, 3, 5, 10),
    "vect__ngram_range": ((1, 1), (1, 2)),  # unigrams or bigrams
    "vect__norm": ("l1", "l2"),
    "clf__alpha": np.logspace(-6, 6, 13),
}

# %%
# In this case `n_iter=40` is not an exhaustive search of the hyperparameters'
# grid. In practice it would be interesting to increase the parameter `n_iter
# to get a more informative analysis. The consequent increase in computing time
# can be handled by increasing the number of CPUs via the `n_jobs` parameter.
```